main.js

```diff
@@ -87,49 +121,39 @@ function cardStorageRefresh() {
     var jsonObject = JSON.stringify(userCard);
     localStorage.setItem(userCard.id, jsonObject);
   }
-}
+};
 
 function enableButton() {
   var titleValue = titleInput.value;
   var bodyValue = bodyInput.value;
-
   if (titleValue !== "" && bodyValue !== "") {
     submit.disabled = false;
     } if (titleValue === "" || bodyValue === "") {
       submit.disabled = true;
   }
 };
 
-function toggleStar(event) {
-  var id = event.target.parentNode.parentNode.parentNode.id;
-
+function toggleStar(passedEvent) {
+  var id = passedEvent.parentNode.parentNode.parentNode.dataset.id;
   for (var i = 0; i < allCards.length; i++) {
     if(allCards[i].id.toString() === id) {
       allCards[i].starred = !allCards[i].starred;
-    }
-  }
+    };
+  };
   cardStorageRefresh();
-}
+};
 
-buttonOpen.addEventListener('click', menuDropdown);
 function menuDropdown() {
   var hide = document.querySelector('.hidden-section');
   navSection.classList.toggle('show');
   buttonOpen.classList.toggle('nav-button-close');
   hide.classList.toggle('hide');
+};
 
-}
-
-function cardRemove(event) {
+function cardRemove(passedEvent) {
   allCards = allCards.filter(allCards => {
```

Comment:
naming the callback param allCards is deceptive here, as it is a single card during each iteration of the filter. Also, naming it the same thing as the variable is generally a no go